src/convertResults.ts

```diff
@@ -0,0 +1,25 @@
+import { CLIEngine, Linter } from "eslint"
+import { CodacyResult } from "./model/CodacyResult"
+import { flatMap } from "lodash"
+import { patternIdToCodacy } from "./model/Patterns"
+
+export function convertResults(report: CLIEngine.LintReport): CodacyResult[] {
+  return flatMap(report.results, result => {
+    const filename = result.filePath
+    const pairs = result.messages.map(m => [m.ruleId, m]).filter(t => t[0]) as [
```

Comment:
is this operation safe? can you explain the rationale behind the `[0]` ?

Reply:
I map first the messages to a [tuple](https://www.typescriptlang.org/docs/handbook/basic-types.html#tuple).
Then I filter them by `ruleId` (to check that ruleId is defined, since it is `string | undefined`).
It is not checked by the compiler, so you can put whatever number you want.
I'm changing it to `.filter(([ruleId, _m]) => ruleId)`. Yet not checked at compile time (you can put whatever number of elements in the tuple matching) but, at least, more readable